editor/src/components/canvas/canvas-strategies/strategies/draw-to-insert-meta-meta-strategy.ts

```diff
@@ -0,0 +1,27 @@
+import { MetadataUtils } from '../../../../core/model/element-metadata-utils'
+import { stripNulls } from '../../../../core/shared/array-utils'
+import { findElementPathUnderInteractionPoint, type MetaCanvasStrategy } from '../canvas-strategies'
+import type { CustomStrategyState, InteractionCanvasState } from '../canvas-strategy-types'
+import type { InteractionSession } from '../interaction-state'
+import { drawToInsertMetaStrategy } from './draw-to-insert-metastrategy'
+import { gridDrawToInsertStrategy } from './grid-draw-to-insert-strategy'
+
+export const drawToInsertMetaMetaStrategy: MetaCanvasStrategy = (
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+  customStrategyState: CustomStrategyState,
+) => {
+  const targetParent = findElementPathUnderInteractionPoint(canvasState, interactionSession)
+
+  if (
+    MetadataUtils.isGridLayoutedContainer(
+      MetadataUtils.findElementByElementPath(canvasState.startingMetadata, targetParent),
+    )
+  ) {
+    return stripNulls([
+      gridDrawToInsertStrategy(canvasState, interactionSession, customStrategyState),
+    ])
+  }
+
+  return drawToInsertMetaStrategy(canvasState, interactionSession, customStrategyState)
+}
```

Comment:
This metametastrategy looks so simple, did you think of just putting this at the beginning of drawToInsertMetaStrategy? It looks unnecessary to me to add another level of meta strategy here.

Reply:
good idea, I did it here https://github.com/concrete-utopia/utopia/pull/6147/commits/cfc3a7259a307657b9bf53dde0c55b8100a713c3